app/controllers/follows_controller.rb

```diff
@@ -30,16 +30,25 @@ def create
                  else
                    User.find(params[:followable_id])
                  end
+
     need_notification = Follow.need_new_follower_notification_for?(followable.class.name)
+    rate_limiter = RateLimitChecker.new(current_user)
+
     @result = if params[:verb] == "unfollow"
-                follow = current_user.stop_following(followable)
-                Notification.send_new_follower_notification_without_delay(follow, true) if need_notification
-                "unfollowed"
+                unfollow(followable, need_notification: need_notification)
               else
-                follow = current_user.follow(followable)
-                Notification.send_new_follower_notification(follow) if need_notification
-                "followed"
+                begin
+                  raise RateLimitChecker::DailyFollowAccountLimitReached if rate_limiter.limit_by_action("follow_account")
+
+                  follow(followable, need_notification: need_notification)
+                rescue RateLimitChecker::DailyFollowAccountLimitReached
+                  respond_to do |format|
+                    format.json { render json: { error: "Daily account follow limit reached!" } }
+                  end
```

Comment:
do we need the `respond_to` here? The method already returns only JSON...

BTW, here you should add a HTTP status error. There's a beautiful HTTP error called [Too Many Requests](https://httpstatuses.com/429) that I think it's apt here. You can add it like this: `render json: { error: "", status: :too_many_requests }`